emcc.py

```diff
@@ -1761,9 +1761,25 @@ def include_and_export(name):
         '_asan_c_store_f', '_asan_c_store_d',
       ]
 
-      shared.Settings.GLOBAL_BASE = shared.Settings.ASAN_SHADOW_SIZE
-      shared.Settings.INITIAL_MEMORY += shared.Settings.ASAN_SHADOW_SIZE
-      assert shared.Settings.INITIAL_MEMORY < 2**32
+      if shared.Settings.ASAN_SHADOW_SIZE != -1:
+        diagnostics.warning('emcc', 'ASAN_SHADOW_SIZE is ignored and will be removed in a future release')
+
+      max_mem = shared.Settings.INITIAL_MEMORY
+      if shared.Settings.ALLOW_MEMORY_GROWTH:
+        max_mem = shared.Settings.MAXIMUM_MEMORY
+        if max_mem < 0:
+          exit_with_error('ASan requires a finite MAXIMUM_MEMORY')
+
+      shadow_size = max_mem // 8
+      shared.Settings.GLOBAL_BASE = shadow_size
+      shared.Settings.INITIAL_MEMORY += shadow_size
```

Comment:
Should this be `shared.Settings.INITIAL_MEMORY += shadow_size - shared.Settings.GLOBAL_BASE` (before you clobber GLOBAL_BASE that is)..  normally the INITIAL_MEMORY includes the unused first 1024 bytes doesn't it?

Reply:
Oh, that makes sense. I wasn't really sure what GLOBAL_BASE was.

Reply:
Oh wait, I think that won't work because then INITIAL_MEMORY wouldn't be wasm-page aligned.